CodeU_Assignement_1/src/codeu_assignement_1/main.java

```diff
@@ -0,0 +1,278 @@
+        // Gets the k-th location out-of the k-th to the last location
+        int k = l.size() - k2last - 1;
+        if (k < 0 || k > l.size()) 
+            throw new IndexOutOfBoundsException();
+    
+        return l.get(k);
+    } // End Q2
+    
+    // Some testings to the algorithms
+    static void run_tests_Q1()
+    {
+       String s1 = null;
+       String s2 = null;
+       System.out.println("Q1: (true) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+       s1 = "";
+       System.out.println("Q1: (false) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+       s2 = "";
+       System.out.println("Q1: (true) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+       s2 = "a";
+       System.out.println("Q1: (false) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+       s1 = null;
+       System.out.println("Q1: (false) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+       s1 = "";
+       System.out.println("Q1: (false) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+       s1 = "a";
+       System.out.println("Q1: (true) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+       s2 = "aa";
+       System.out.println("Q1: (false) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+       s1 = "aaa";
+       System.out.println("Q1: (false) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+       s2 = "";
+       System.out.println("Q1: (false) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+       s1=" ";
+       System.out.println("Q1: (false) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+       s1="abc";
+       s2="bcd";
+       System.out.println("Q1: (false) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+             
+        s1="Listen";
+        s2="Silent";
+        System.out.println("Q1: (true) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+       
+        s1="Triangle";
+        s2="Integral";
+        System.out.println("Q1: (true) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+        s1="Apple";
+        s2="Pabble";
+        System.out.println("Q1: (false) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+    }

```

Comment:
Yes, it is a simple tests - Assignment 1 isn't really too complicated. I can give unit-test a try in Assignment 2